cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -386,33 +393,59 @@ ignoringSeverity :: Tracer IO a -> Tracer IO (WithSeverity a)
 ignoringSeverity tr = Tracer $ \(WithSeverity _ ev) -> traceWith tr ev
 {-# INLINE ignoringSeverity #-}
 
+countMyBlocks
+  :: forall blk. ()
+  => HasHeader (Header blk)
+  => BlockSupportsMetrics blk
+  => BlockConfig blk
+  -> AF.AnchoredFragment (Header blk)
+  -> Int64
+countMyBlocks blockConfig = fromIntegral . length . AS.filter isMine
+  where isMine :: Header blk -> Bool
+        isMine header = isSelfIssued blockConfig header == IsSelfIssued
```

Comment:
One last caveat: @disassembler on Slack asked which identity the `isSelfIssued` is using to determine "self". It's the cold key. I'm not sure if that's what he expected/wanted/needed; we haven't continued the discussion.

Reply:
(we resolved this: _the public half of_ cold key we're using is fine)